compiler/rustc_target/src/abi/mod.rs

```diff
@@ -1355,6 +1364,14 @@ pub struct PointeeInfo {
     pub address_space: AddressSpace,
 }
 
+/// Used in `might_permit_raw_init` to indicate the kind of initialisation
+/// that is checked to be valid
+#[derive(Copy, Clone, Debug)]
+pub enum ValidAs {
```

Comment:
:+1: for making this an enum. However, I don't quite understand `ValidAs` as the name. What about `InitKind`?

Reply:
It was along the lines of "is this type valid as zeroed, or valid as uninit".

In any case, I changed it, I'm not too bothered what it's called.